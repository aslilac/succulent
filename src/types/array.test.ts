/// <reference types="jest" />

import { check, is, $array, $number, $object, $string, $undefined } from "../index";

test("$array", () => {
	const numArray = $array($number);
	expect(is([], numArray)).toBe(true);
	expect(is([1], numArray)).toBe(true);
	expect(is([1, 2, 3, 4, 5], numArray)).toBe(true);
	expect(is([1, 2, 3, 4, 5, undefined], numArray)).toBe(false);

	expect(() =>
		check([1, 2, 3, 4, 5, undefined], numArray),
	).toThrowErrorMatchingSnapshot();

	const undefinedArray = $array($undefined);
	expect(is(new Array(100), undefinedArray)).toBe(true);
	expect(is(new Array(100), numArray)).toBe(false);

	const $Thing = $object({ name: $string, count: $number });

	expect(() => check(false, $array($Thing))).toThrowErrorMatchingSnapshot();
	expect(() => check([false], $array($Thing))).toThrowErrorMatchingSnapshot();
});
