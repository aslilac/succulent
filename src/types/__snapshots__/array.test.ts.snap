// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`$array 1`] = `
"Expected number[], got 1,2,3,4,5,
  Expected number, got undefined"
`;

exports[`$array 2`] = `"Expected Array<{name: string, count: number}>, got false"`;

exports[`$array 3`] = `
"Expected Array<{name: string, count: number}>, got false
  Expected {name: string, count: number}, got false
    Expected object, got false"
`;
